verified_boot_app.md

```diff
@@ -57,12 +173,112 @@ Src: https://github.com/Kicksecure/kicksecure-wiki-backup/blob/6900fdfbdea8693fd
+* Must be a feature that the firmware developer considers potentially commercially beneficial (as in customers would buy motherboards with such freedom-preserving firmware): yes
+-->
+
+3.1. The application is RECOMMENDED be integrated into the system's firmware
+     image.
+
+3.2. The application MAY be launched from a predefined location on EFI System
+     Partition (ESP). The application MUST be signed with a key or possess a hash
+     trusted by the firmware if this is done.
+
+3.3. The application MUST be run on the very first power on of the system and
+     on every `EFI_BOOT_MODE` set to `BOOT_WITH_DEFAULT_SETTINGS` or
+     `BOOT_WITH_MFG_MODE_SETTINGS` per UEFI Platform Initialization
+     Specification.
+
+<!--
+* Technical high-level overview:
+** Power on -> find list of candidate bootloaders (or let user choose bootloader to load via boot menu, as usual)
+** Loop through the list trying to verify and boot each one
+** Verify bootloader signature against in-firmware trusted Secure Boot database
+** If verification passes, continue boot
+** EDIT: If verification fails, pass control to the application
+-->
+
+3.4. The firmware MAY run the application if the bootloader image verification
+     fails and the firmware is unable to boot the operating system.
+
+<!--
+There will be no enforcement of image verification if permissive policies or
+bugs like MSI inSecure Boot will be present in the firmware.
+-->
+
+3.5. The firmware SHOULD NOT implement insecure image verification policies
+     allowing to bypass signature verification when UEFI Secure Boot is
+     enabled.
+
+## Usecase examples
+
+<!--
 * Technical high-level examples:
 ** Power on -> firmware detects Microsoft Windows (bootloader) is installed -> show the key to the user -> user confirms or denies the key -> continue boot (if confirmed, otherwise go back to firmware menu or something)
 ** Power on -> firmware detects a "standard" Linux distribution such as Debian without a key in the well-known location -> show default Microsoft signing key for shim to the user -> user confirms or denies the key -> continue boot (if confirmed, otherwise go back to firmware menu or something)
 ** Power on -> firmware detects an "augmented" Linux distribution such as Kicksecure with a key database in the well-known location -> show key from key database to the user -> user confirms or denies the key -> continue boot (if confirmed, otherwise go back to firmware menu or something)
 ** User disables Secure Boot -> as usual
-* Additional notes:
-** To handle the situation where multiple shims are installed, the firmware may need the ability to find multiple candidate bootloaders in the removable media path (or any bootloader path)
+-->
 
 -->
+
+### Initial configuration
+
+<!-- TODO: diagrams? -->
+
+1. User powers on the platform.
+2. Firmware performs boot media enumeration.
+3. Firmware detects initial boot or boot with default settings and launches an
+   application.
```

Comment:
We should focus on initial boot and then design the following boot and other potential error paths, like recovery/fallback. Mixing seems to be complex.

Reply:
I just put the use case written down in the wiki in a more technical way.

Reply:
That is fine, but what help would it provide here? I thought we placed it here to refine it and make it something we will implement, not as an ephemeral checkbox, just to clean up what was on the wiki.